### ISSUE TITLE
How can we simplify and streamline tag setup?

### ISSUE THREAD

**Issue by hburgund** [roundware/roundware-server#191]
Could we remove UI mappings by integrating into MasterUI and tags directly?

This is a very preliminary idea to simplify the tag setup; I'm not sure it will be sensible to do without undesirably limiting flexibility, but I wanted to capture thoughts on the topic for eventual evaluation and discussion.

This would require limiting each tag to a single MasterUI/Project, but that's basically what is done anyway as best practice.  It would also not allow for different ordering of tags per mode, which isn't ideal, but probably isn't a huge deal.

UI Mappings has five fields:

```
Master ui
Index
Tag
Active
Default
```

If we were to migrate the `index` `active` and `default` fields from `uimapping` to `tag` and add a `project` field to `tag` as well, I think that would do the trick.  Since `masterui` already has `project` and `tag_category`, that would effectively assign tags to the proper `masterui` indirectly.  But is that really any easier?

Perhaps we just need to try again to improve the admin UI for tag setup as that's still complicated despite the work done on this master ui admin screen:

http://localhost/rw/masteruis/setup_tag_ui/

Lots of good work was done here, but I still don't really use it because there are some issues that make it not totally dependable: #66, #67 and #71.

**Comment by eosrei** [comment]
Note: #200 removes the UIMode Table.

**Comment by eosrei** [comment]
@hburgund Should tags be project specific? Seems as though they should.

**Comment by hburgund** [comment]
Yeah, I think they should be.  I originally kept them as not project-specific as I thought we would re-use ones between projects, but after having this option for numerous years, it turns out that there are significant downsides to sharing tags between projects, so I have been keeping them project-specific even though I don't need to.
